echo_server_reactor.py

```diff
@@ -0,0 +1,69 @@
+import sys
+import socket
+import select
+# from _typeshed import FileDescriptorLike
+from collections.abc import Callable, Iterable
+
+SIZE = 1024
+MINUTE = 60
+NUM_LISTEN = 5
+
+
+class Reactor(object):
+    def __init__(self):
+        self.handler = select.poll()
+        self.fd_to_action: dict[int: (Callable, Iterable)] = {}
+
+    def register(self, file_obj: socket.socket, event_mask: int, func: Callable, args: Iterable = ()):
+        self.handler.register(file_obj, event_mask)
+        self.fd_to_action[file_obj.fileno()] = (func, args)
+
+    def unregister(self, file_obj: socket.socket):
+        self.handler.unregister(file_obj)
+
+    def handle_events(self):
+        while True:
+            events = self.handler.poll()
+            for key, mask in events:
```

Comment:
we are not using "mask" so it is better to be explicit about it 

```suggestion
            for key, _ in events:
```